modules/openapi/component-protocol/scenarios/auto-test-plan-detail/components/executeTaskTable/render.go

```diff
@@ -488,6 +499,28 @@ func (a *ExecuteTaskTable) setData(pipeline *apistructs.PipelineDetailDTO) error
 	return nil
 }
 
+// getCostTime the format of time is "00:00:00"
+// id is not end status or err return "-"
+// if task is snippet return the costTime of pipeline_bases,
+// if not return pipeline_tasks's
+func (a *ExecuteTaskTable) getCostTime(task apistructs.PipelineTaskDTO) string {
+	if !task.Status.IsEndStatus() {
+		return "-"
+	}
+	if !task.IsSnippet {
+		return time.Unix(task.CostTimeSec, 0).In(time.UTC).Format("15:04:05")
+	}
+	if task.SnippetPipelineID == nil {
+		return "-"
+	}
+
+	rsp, err := a.CtxBdl.Bdl.GetPipeline(*task.SnippetPipelineID)
```

Comment:
snippet task can also use task's costTimeSec.

Reply:
You have to care about performance issue.

Reply:
the costTimeSec  of snippet task is always -1

Reply:
now update snippet pipeline task time when reconciler done